src/libs/fileDownload/FileUtils.ts

```diff
@@ -274,56 +274,138 @@ function validateImageForCorruption(file: FileObject): Promise<{width: number; h
+                        resolve(result);
+                    } catch (e) {
+                        resolve(false);
+                    }
+                };
 
                 stream.onData((chunk: string | number[]) => {
-                    const chunkStr = Array.isArray(chunk) ? String.fromCharCode(...chunk) : chunk;
-                    base64Chunk += chunkStr;
+                    if (hasEnoughData) {
+                        return;
+                    }
+
+                    try {
+                        let chunkStr: string;
+                        if (Array.isArray(chunk)) {
+                            chunkStr = chunk.map((code) => String.fromCharCode(code)).join('');
+                        } else {
+                            chunkStr = chunk;
+                        }
+                        base64Data += chunkStr;
+
+                        const decodedByteCount = Math.floor((base64Data.length * 3) / 4);
+                        if (decodedByteCount >= MAGIC_BYTES_NEEDED) {
+                            hasEnoughData = true;
+                            processData();
+                        }
+                    } catch (e) {
+                        if (!hasEnoughData) {
+                            hasEnoughData = true;
+                            resolve(false);
+                        }
+                    }
                 });
 
-                stream.onError((error: Error) => {
-                    reject(error);
+                stream.onError(() => {
+                    if (hasEnoughData) {
+                        return;
+                    }
+                    hasEnoughData = true;
+                    resolve(false);
                 });
 
                 stream.onEnd(() => {
-                    resolve(base64Chunk);
+                    if (hasEnoughData) {
+                        return;
+                    }
+                    hasEnoughData = true;
+                    processData();
                 });
-            });
-        })
-        .then((base64Data: string) => {
-            const binary = Buffer.from(base64Data, 'base64').toString('binary');
-
-            const bytes = new Uint8Array(binary.length);
-            for (let i = 0; i < binary.length; i++) {
-                bytes[i] = binary.charCodeAt(i);
-            }
-
-            const startIndex = Math.min(4, bytes.length);
-            const endIndex = Math.min(16, bytes.length);
-            const view = bytes.subarray(startIndex, endIndex);
-
-            const hex = Array.from(view)
-                .map((b) => b.toString(16).padStart(2, '0'))
-                .join('');
 
-            return hex;
-        })
-        .then((hexSignature: string) => {
-            return formatSignatures.some((signature) => hexSignature.startsWith(signature));
-        })
-        .catch(() => {
-            return false;
-        });
+                stream.open();
```

Comment:
I think we should also close the stream when the task has done to prevent memory leak.

Reply:
Yeah I think you're right